internal/server/authn/oidc.go

```diff
@@ -48,14 +47,14 @@ func (a *oidcAuthn) Authenticate(ctx context.Context, db *data.Transaction, requ
 		return AuthenticatedIdentity{}, fmt.Errorf("exhange code for tokens: %w", err)
 	}
 
-	if len(provider.AllowedDomains) > 0 {
+	if len(a.AllowedLoginDomains) > 0 {
```

Comment:
Just to confirm, do we verify somewhere that this is the "social" provider and not a random 3rd party IDP?

Reply:
The check for social login is in #3657, I removed this logic and added a TODO that I will resolve in that PR instead